### ISSUE TITLE
Uncaught TypeError: target.methods.generateAndInjectStyles is not a function

### ISSUE THREAD

**Issue by maratfakhreev** [styled-components/vue-styled-components#95]
Hey. In my project I've decided update the version of vue-styled-components from 1.4.4 to 1.4.13 and the app has broken. The error is next:

```
vue-styled-components.es.js:3058 Uncaught TypeError: target.methods.generateAndInjectStyles is not a function
    at isStyledComponent (vue-styled-components.es.js:3058)
    at isValidElementType (vue-styled-components.es.js:3062)
    at styled (vue-styled-components.es.js:3069)
    at Object.<anonymous> (index.jsx:78)
    at __webpack_require__ (bootstrap 1225f84d72bb390aced3:19)
    at Object.<anonymous> (index.jsx:4)
    at __webpack_require__ (bootstrap 1225f84d72bb390aced3:19)
    at Object.<anonymous> (reports.png:1)
    at __webpack_require__ (bootstrap 1225f84d72bb390aced3:19)
    at Object.<anonymous> (router.js:1)
```

In index.jsx:78 I'm overriding Vue component (SetPopover) which has it's own methods section:
```
const SortPopover = styled(SetPopover)`
  right: 0;
`;
```

So in `vue-styled-components.es.js:3058` we can see next method:
```
function isStyledComponent(target) {
  return target && target.methods && typeof target.methods.generateAndInjectStyles() === 'string';
}
```
and as far as I understand the problem is b/c `target.methods` doesn't have `generateAndInjectStyles` method, it has their own methods from Vue component. So I've tried to add additional checking, but after that my application has broken in another place, when I've tried to override VueSlider npm package.

Have you changed a library api? It seems that it is now provides the incorrect behavior of overriding components that not a styled.

**Comment by JessicaSachs** [comment]
This bug is a regression. It only happens with components that have methods. I suggest adding a safety check... I'll put up a PR.

**Comment by liqueflies** [comment]
Fixed in #96.
